File: Autos-stock-manager-services/src/main/java/org/autostock/enums/TypeDocument.java
package org.autostock.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum TypeDocument {

    FACTURE("Facture", "INVOICE"),
    IMMATRICULATION("Certificat d'immatriculation", "REGISTRATION"),
    INSPECTION("Rapport d'inspection", "INSPECTION"),
    CONTRAT("Contrat de vente", "CONTRACT"),
    AUTRE("Autre document", "OTHER");

    private final String label;
    private final String value;

    TypeDocument(String label, String value) {
        this.label = label;
        this.value = value;
    }

    public String getLabel() {
        return label;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    public static TypeDocument fromValue(String value) {
        for (TypeDocument t : values()) {
            if (t.value.equalsIgnoreCase(value)) {
                return t;
            }
        }
        throw new IllegalArgumentException("Type de document inconnu : " + value);
    }
}

File: autos-stock-client/src/app/models/enums/TypeDocument.ts
// Noms de constantes de org.autostock.enums.TypeDocument : c'est ce que
// l'API renvoie et ce que TypeDocument.valueOf() attend a l'ecriture.
export type TypeDocument =
  | 'FACTURE'
  | 'CARFAX'
  | 'PHOTO'
  | 'IMMATRICULATION'
  | 'INSPECTION'
  | 'CONTRAT'
  | 'RECU_PAIEMENT'
  | 'AUTRE';
